netlogo-gui/src/main/app/EditorFactory.scala

```diff
@@ -2,32 +2,39 @@
 
 package org.nlogo.app
 
-import javax.swing.KeyStroke
+import javax.swing.{Action, KeyStroke}
 import javax.swing.text.TextAction
 
 import org.nlogo.api.CompilerServices
 import org.nlogo.core.I18N
-import org.nlogo.ide.{AutoSuggestAction, CodeCompletionPopup}
+import org.nlogo.ide.{AutoSuggestAction, CodeCompletionPopup, ShowUsageBox, ShowUsageBoxAction}
+import org.nlogo.window.CodeEditor
+
 import collection.JavaConversions._
 
 class EditorFactory(compiler: CompilerServices) extends org.nlogo.window.EditorFactory {
-  def newEditor(cols: Int, rows: Int, disableFocusTraversal: Boolean) =
-    newEditor(cols, rows, disableFocusTraversal, null, false)
+  def newEditor(cols: Int, rows: Int, enableFocusTraversal: Boolean): CodeEditor =
+    newEditor(cols, rows, enableFocusTraversal, null, false)
+
   def newEditor(cols: Int,
                 rows: Int,
-                disableFocusTraversal: Boolean,
+                enableFocusTraversal: Boolean,
                 listener: java.awt.event.TextListener,
-                isApp: Boolean) =
+                isApp: Boolean): CodeEditor =
   {
     val font = new java.awt.Font(org.nlogo.awt.Fonts.platformMonospacedFont,
                                  java.awt.Font.PLAIN, 12)
     val colorizer = new org.nlogo.window.EditorColorizer(compiler)
     val codeCompletionPopup = new CodeCompletionPopup()
-    val actionMap = mapAsJavaMap(Map(KeyStroke.getKeyStroke(java.awt.event.KeyEvent.VK_SPACE, java.awt.event.InputEvent.CTRL_DOWN_MASK)
-      -> new AutoSuggestAction("auto-suggest", codeCompletionPopup))).asInstanceOf[java.util.Map[KeyStroke, TextAction]]
+    val showUsageBox = new ShowUsageBox()
+    val actions = Seq[Action](new ShowUsageBoxAction(showUsageBox))
+    val actionMap = Map(
+      KeyStroke.getKeyStroke(java.awt.event.KeyEvent.VK_SPACE, java.awt.event.InputEvent.CTRL_DOWN_MASK)
+      -> new AutoSuggestAction("auto-suggest", codeCompletionPopup))
```

Comment:
Minor code style tweak: it would be more typical style to have `->` at the end of line 32 than at the beginning of line 33.